tests/unit/jobs/test_views.py

```diff
@@ -723,3 +723,89 @@ def test_change_submission_data_and_visibility_when_submission_doesnt_exist(self
+                "when_made_public": "{0}{1}".format(self.submission.when_made_public.isoformat(),
+                                                    'Z').replace("+00:00", ""),
+            }
+
+        self.client.force_authenticate(user=self.user)
+
+        self.challenge.participant_teams.add(self.participant_team)
+        response = self.client.get(self.url)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_submission_by_pk_when_user_created_the_submission(self):
+        self.url = reverse_lazy('jobs:get_submission_by_pk',
+                                kwargs={'submission_id': self.submission.id})
+        expected = {
+                'id': self.submission.id,
+                'participant_team': self.submission.participant_team.pk,
+                'participant_team_name': self.submission.participant_team.team_name,
+                'execution_time': self.submission.execution_time,
+                'challenge_phase': self.submission.challenge_phase.pk,
+                'created_by': self.submission.created_by.pk,
+                'status': self.submission.status,
+                'input_file': "http://testserver%s" % (self.submission.input_file.url),
+                'method_name': self.submission.method_name,
+                'method_description': self.submission.method_description,
+                'project_url': self.submission.project_url,
+                'publication_url': self.submission.publication_url,
+                'stdout_file': None,
+                'stderr_file': None,
+                'submission_result_file': None,
+                "submitted_at": "{0}{1}".format(self.submission.submitted_at.isoformat(), 'Z').replace("+00:00", ""),
+                "is_public": self.submission.is_public,
+                "when_made_public": "{0}{1}".format(self.submission.when_made_public.isoformat(),
+                                                    'Z').replace("+00:00", ""),
+            }
+
+        self.client.force_authenticate(user=self.submission.created_by)
+
+        self.challenge.participant_teams.add(self.participant_team)
+        response = self.client.get(self.url)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_get_submission_by_pk_when_user_is_neither_challenge_host_nor_submission_owner(self):
+        self.url = reverse_lazy('jobs:get_submission_by_pk',
+                                kwargs={'submission_id': self.submission.id})
+
+        expected = {'error': 'Sorry, you are not authorized to access this submission.'}
+
+        self.challenge.participant_teams.add(self.participant_team)
+        response = self.client.get(self.url)
+        self.assertEqual(response.data, expected)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_get_submission_by_pk_when_submission_doesnt_exist(self):
```

Comment:
Please try to write the test cases as per the code. The test case for the code written first must come first.